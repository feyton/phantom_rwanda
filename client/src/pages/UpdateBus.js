import React from 'react';
import { useLocation } from 'react-router-dom';
import { toast } from 'react-toastify';
import Buses from '../components/forms/Buses.js';
import { axiosBase as axios } from '../utils/Api.js';

const UpdateBus = () => {
  const location = useLocation();
  const { id, busType, plateNumber, company, seats } = location.state;
  /* istanbul ignore next */
  const updateBusInDB = async (newBusInfo) => {
    try {
      await axios.put(`/buses/${id}`, newBusInfo);
      toast('Bus updated successfully', { type: 'success' });
      return;
    } catch (error) {
      return Promise.reject(error);
    }
  };
  return (
    <Buses
      formTitle="Update bus"
      formAction={updateBusInDB}
      busType={busType}
      company={company}
      seats={seats}
      plateNumber={plateNumber}
    />
  );
};

export default UpdateBus;
