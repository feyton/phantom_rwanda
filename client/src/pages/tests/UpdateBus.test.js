import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import UpdateBus from '../UpdateBus.js';

const mockUseLocationValue = {
  state: {
    busType: 'Minibus',
    company: 'Virunga_express',
    seats: '30',
    plateNumber: 'TAN123',
    id: 3
  }
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  })
}));
describe('RegisterBus component test', () => {
  it('it should render update bus page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <UpdateBus />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
