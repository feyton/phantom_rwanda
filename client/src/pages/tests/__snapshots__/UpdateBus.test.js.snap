// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterBus component test it should render update bus page 1`] = `
<div
  className="px-8 py-4 my-auto font-raleway md:py-8"
>
  <form
    className="py-8 shadow-main px-4 flex flex-col w-full md:max-w-lg md:mx-auto"
    onSubmit={[Function]}
  >
    <h1
      className="font-extrabold text-2xl pb-8"
    >
      Update bus
    </h1>
    <label
      className="font-bold mb-2"
      htmlFor="busType"
    >
      Bus type
    </label>
    <select
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      defaultValue="Minibus"
      id="busType"
      name="busType"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        hidden={true}
        value=""
      >
        Select bus type
      </option>
      <option
        value="Coaster"
      >
        Coaster
      </option>
      <option
        value="Minibus"
      >
        Minibus
      </option>
      <option
        value="Big Bus"
      >
        Big Bus
      </option>
    </select>
    <p
      className="text-rose-500 mb-2"
    />
    <label
      className="font-bold mb-2"
      htmlFor="busType"
    >
      Company
    </label>
    <select
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      defaultValue="Virunga_express"
      id="company"
      name="company"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        hidden={true}
        value=""
      >
        Select company
      </option>
      <option
        value="KBS"
      >
        Kigali Bus Service
      </option>
      <option
        value="Royal express"
      >
        Royal Express
      </option>
      <option
        value="Virunga express"
      >
        Virunga Express
      </option>
    </select>
    <p
      className="text-rose-500 mb-2"
    />
    <label
      className="font-bold mb-2"
      htmlFor="seats"
    >
      Seats
    </label>
    <input
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      defaultValue="30"
      name="seats"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter number of seats"
      type="number"
    />
    <p
      className="text-rose-500 mb-2"
    />
    <label
      className="font-bold mb-2"
      htmlFor="seats"
    >
      Plate number
    </label>
    <input
      className="rounded-sm px-3 py-3 mb-2 bg-[#EFEFEF]"
      defaultValue="TAN123"
      name="plateNumber"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter plate number"
      type="text"
    />
    <p
      className="text-rose-500 mb-2"
    />
    <div
      className="self-center"
    >
      <div>
        <button
          className="bg-primary text-sm text-white py-2 px-3 mt-4 rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          Update bus
        </button>
      </div>
    </div>
  </form>
</div>
`;
